src/hardware/input/intel8042.cpp

```diff
@@ -371,6 +371,61 @@ static uint8_t get_translated(const uint8_t byte)
 	return translation_table[byte];
 }
 
+// ***************************************************************************
+// Port disable workaround
+// ***************************************************************************
+
+// TODO: This part is a workaround for 'Ultima VIII: Pagan' problem with
+// non-working keyboard. The proper fix would require larger rework:
+// - it seems that even if keyboard is disabled, the 8042 should fire the
+//   interrupt
+// - testing with 86Box revealed that command 0xad (disable keyboard port) and
+//   0xae (enable port) are being sent by BIOS with each and every key press;
+//   our internal BIOS does not work this way
+//
+// Ignoring command 0xad totally might be risky, as Windows 3.11 for Workgroups
+// disables/reenables the keyboard port during startup and shutdown, issuing
+// controller commands in between; if something disturbs it, we might run into
+// a problem with keyboard/mouse not responding at all.
+//
+// This code re-enables the keyboard after certain time - unless there is
+// some command currently being issued to the controller.
+
+static bool kbd_disabled_timer_running = false;
+static bool kbd_disabled_timer_expired = true;
+
+constexpr double kbd_disabled_duration_ms = port_delay_ms * 2;
```

Comment:
UpperCamelCaseConstants

Reply:
Updated in a separate commit, cleans up many constant names.